datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,348 @@
+    main_grid = index_page.find('.//div[@id="main-grid"]')
+    if main_grid is not None:
+        for link in main_grid.xpath(".//a/@href"):
+            if link.startswith("https:"):
+                if data_type == "person":
+                    crawl_person(context, link)
+                elif data_type == "company":
+                    crawl_company(context, link)
+                if data_type == "vessel":
+                    crawl_vessel(context, link)
+
+
+def crawl_vessel(context: Context, link):
+    detail_page = context.fetch_html(link, cache_days=3)
+    details_container = detail_page.find(".//main")
+    data: dict[str, str] = {}
+
+    xpath_definitions = [
+        (
+            "main_info_rows",
+            ".//div[contains(@class,'tools-spec')]/div[contains(@class, 'row')]",
+        ),
+        (
+            "justification_rows",
+            ".//div[contains(@class, 'tools-frame')]/div[contains(@class, 'mb-3')]",
+        ),
+        (
+            "additional_info_rows",
+            ".//div[contains(@class, 'tools-frame')]//div[@class='mb-3' or contains(@class, 'js_visibility')]",
+        ),
+    ]
+    for name, xpath_expr in xpath_definitions:
+        rows = details_container.xpath(xpath_expr)
+        for row in rows:
+            divs = row.findall("div")
+            if len(divs) == 2:
+                label_elem, value_elem = divs
+                if "yellow" in value_elem.get("class"):
+                    label, value = extract_label_value_pair(
+                        label_elem, value_elem, data
+                    )
+                    data[label] = value
+
+    web_resources = []
+    web_links = details_container.xpath(
+        ".//div[contains(@class, 'tools-frame')]//a[contains(@class, 'long-text yellow')]"
+    )
+    for raw_link in web_links:
+        link_href = raw_link.get("href", "").strip()
+        web_resources.append(link_href)
+
+    name = data.pop("Vessel name (international according to IMO)")
+    type = data.pop("Vessel Type")
+    imo_num = data.pop("IMO")
+
+    vessel = context.make("Vessel")
+    vessel.id = context.make_id(name, imo_num)
+    vessel.add("name", name)
+    vessel.add("imoNumber", imo_num)
+    vessel.add("type", type)
+    vessel.add("description", " ".join(data.pop("Vessel information")))
+    vessel.add("callSign", data.pop("Call sign"))
+    vessel.add("flag", data.pop("Flag (Current)"))
+    vessel.add("mmsi", data.pop("MMSI"))
+    vessel.add("buildDate", data.pop("Build year"))
+    vessel.add("sourceUrl", web_resources)
+    vessel.add("keywords", data.pop("Category"))
+    for pr_name in h.multi_split(data.pop("Former ship names"), [" / "]):
+        vessel.add("previousName", pr_name)
+    for past_flag in h.multi_split(data.pop("Flags (former)"), [" /"]):
+        vessel.add("pastFlags", past_flag)
+    vessel.add("topics", "poi")
+
+    sanction = h.make_sanction(context, vessel)
+    sanction.add("country", data.pop("Sanctions", None))
```

Comment:
this would have the effect of saying the sanction authority `Defence Intelligence of Ukraine` has sanctioned the vessel in all the countries where other countries' sanctions authorities have sanctioned the vessel. So let's not add the other sanctioning countries here.

Let's add Program to indicate that this is part of the `Vessels, Airports and Ports involved in the transportation of weapons, stolen Ukrainian products and in the circumvention of sanctions` list.
I got that from the heading on the homepage/list page